### DISCUSSION TITLE
NEXT MAJ : proposer en choix les noms de guilde pour la commande guild

### DISCUSSION THREAD

**Discussion by romain22222** [Crownicles/Crownicles#1026] [Nouvelles idées]
faire en sorte de récupérer toutes les noms de guilde de la db, et de proposer ces choix dans le menu déroulant des slash commands

**Comment by Pietagorh** [comment]
Impossible à faire nan ?

**Reply by BastLast** [reply]
On pourrai afficher que 25 guildes

**Comment by Arkitu** [comment]
Je crois qu'il y a un truc avec les interactions de Discord qui permet ça, les autocomplete interaction : https://discord.com/developers/docs/interactions/application-commands#autocomplete
C'est le bot qui pendant que l'utilisateur tape renvoie une liste de résultats potentiels. J'y avais pensé à un moment et je pense que ça serait une bonne idée de l'intégrer

**Reply by romain22222** [reply]
le problème des auto complétions est qu'il y a un nombre de possibilités limité, on ne pas en mettre à l'infini (on est limité à 25 entrées, et il y a actuellement 300+ guildes)

**Reply by BastLast** [reply]
800+

**Reply by Arkitu** [reply]
> le problème des auto complétions est qu'il y a un nombre de possibilités limité, on ne pas en mettre à l'infini (on est limité à 25 entrées, et il y a actuellement 300+ guildes)

Tu es sûr qu'on parle de la même chose ? Je parle pas des `addChoice("choix1", "choix1")` qu'on peut faire à la déclaration des commandes mais des autocomplete qui permettent selon la doc de discord de "retourner dynamiquement des suggestions d'option à un utilisateur en train d'écrire"

**Reply by BastLast** [reply]
WOW ca m'intéresse ça

**Reply by Pietagorh** [reply]
Si ça marche bien, go mettre aussi pour le /test nan ? 👀

**Reply by BastLast** [reply]
Si c'est ben go le mettre partout si l'impact sur les perfs est pas trop méchant parce que ça ressemble à du bcp de requetes sql d'un coup ça
